libcxx/test/support/test_iterators.h

```diff
@@ -267,6 +267,135 @@ template <class It>
 random_access_iterator(It) -> random_access_iterator<It>;
 #endif
 
+// Since C++20, a container iterator type that is random access is also required to support three-way comparison.
```

Comment:
I wonder, would it make sense to guard this with `#if TEST_STD_VER >= 20`? Is there a use-case for this iterator given that we already have a `random_access_iterator`?

Reply:
The (possibly only) use case is making `MinSequenceContainer<T>` a proper sequence container since C++20. An alternative approach IMO would be adding `operator<=>` to `random_access_iterator`, but I don't know whether this is desired.